hana/lib/HANAService.js

```diff
@@ -141,7 +141,9 @@ class HANAService extends SQLService {
     const isSimple = temporary.length + blobs.length + withclause.length === 0
 
     // REVISIT: add prepare options when param:true is used
-    const sqlScript = isLockQuery || isSimple ? sql : this.wrapTemporary(temporary, withclause, blobs)
+    let sqlScript = isLockQuery || isSimple ? sql : this.wrapTemporary(temporary, withclause, blobs)
+    const{ hints } = query.SELECT
+    if (hints) sqlScript += ` WITH HINT (${Array.isArray(hints) ? hints.join(',') : hints})`
```

Comment:
```suggestion
    if (hints) sqlScript += ` WITH HINT (${hints.join(',')})`
```

hints is always an array, isn't it?

Reply:
Yes, the ql `.hints()` API always creates an array. So as long as the property isn't set to something else by hand the check isn't needed.